apps/cli/templates/build.ts

```diff
@@ -0,0 +1,85 @@
+import { readdirSync, lstatSync, readFileSync, writeFileSync } from "node:fs";
+import { join } from "node:path";
+import { type Folder, ProjectType, ensureFileInPath } from "../src/utils";
+
+const TEMPLATES_TO_BUILD = Object.keys(ProjectType);
+
+console.log("Building Templates...");
+const parseFolder = (folderPath: string): Folder | undefined => {
+  const isDirectory = lstatSync(folderPath).isDirectory();
+  if (isDirectory === false) {
+    return;
+  }
+
+  const folderName = folderPath.split("/").pop();
+  if (folderName === undefined) {
+    return;
+  }
+
+  const folder: Folder = {
+    name: folderName,
+    files: [],
+    subFolders: [],
+  };
+
+  const files = readdirSync(folderPath);
+
+  for (let i = 0; i < files.length; i++) {
```

Comment:
2 problems: 
1. we say there should be no single char variables, otherwise people tend to use them where it becomes unreadable
2. why is this not a `for of` ?